packages/picasso.js/src/core/dock-layout/dock-layout.js

```diff
@@ -308,10 +361,19 @@ export default function dockLayout(initialSettings) {
     validateComponent(component);
     docker.removeComponent(component);
 
+    function getReferencedDocks(dock) {
```

Comment:
It's a pure function, there is no need to create this function every time a component is added.

Reply:
fixed!